From a46e381a9007ab552d4551711611cd9d231a9f17 Mon Sep 17 00:00:00 2001
From: Roger Zanoni <rzanoni@igalia.com>
Date: Wed, 25 Oct 2023 15:59:23 -0300
Subject: [PATCH 22/33] [M118-fix] Fix aggregate initialization in trace_log

clang 14 doesn't parenthesized initialization of aggregates, so we are
explicitly defining the needed constructor.

Upstream-Status: Inappropriate
Signed-off-by: Roger Zanoni <rzanoni@igalia.com>
---
 base/trace_event/trace_log.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/base/trace_event/trace_log.h b/base/trace_event/trace_log.h
index 2778526e87b78..3b7e0f0306cf5 100644
--- a/base/trace_event/trace_log.h
+++ b/base/trace_event/trace_log.h
@@ -420,6 +420,12 @@ class BASE_EXPORT TraceLog :
 
 #if BUILDFLAG(USE_PERFETTO_CLIENT_LIBRARY)
   struct TrackEventSession {
+    TrackEventSession(uint32_t internal_instance_index,
+                      perfetto::DataSourceConfig config,
+                      perfetto::BackendType backend_type = perfetto::kUnspecifiedBackend)
+        : internal_instance_index{internal_instance_index}
+        , config{config}
+        , backend_type{backend_type} {}
     uint32_t internal_instance_index;
     perfetto::DataSourceConfig config;
     perfetto::BackendType backend_type = perfetto::kUnspecifiedBackend;
-- 
2.42.1

